### ISSUE TITLE
Documentation points to mathics.net which is hopelessly out-of-date

### ISSUE THREAD

**Issue by TobiSchluter** [mathics/Mathics#762]
Per @wolfv 's request, I'm spining this off #754

The most visible site running mathics is http://www.mathics.net. It is hopelessly out-of-date.

The canonical site seems to be https://mathics.angusgriffith.com/.  This site has broken certificates and a hard-to-remember domain name. Moreover, if you open the documentation on it, you are pointed to the "bad" site, see the screenshot below.

![screen shot 2018-10-14 at 12 53 56](https://user-images.githubusercontent.com/11887541/46913918-58384f80-cfd1-11e8-8a81-dca2af402661.png)

I think there are a number of issues which I vaguely sort in terms of ease-of-addressing (note that I write 'should', but please don't take that as more than a way of expressing potential aciton items, I'm definitely not trying to tell anybody what they *should* do):
- the documentation should not point to mathics.net
- certificates of angusgrifftih.com should be fixed
- a beginner-friendly domain name should be found (actually mathics.org appears available)
- somebody should volunteer to host mathics using that domain (someone very nice :) )
- mathics.net should be reclaimed for the project

It would be important to make sure that there is an easy way for potential users to not run into an old version of mathics when they first try it, because simple bugs (such as my #754 ) are sure to turn potential users away quickly from this very good piece of software.

**Comment by wolfv** [comment]
As mentioned elsewhere, I've set up Mathics in Binder: https://mybinder.org/v2/gh/wolfv/Mathics/binder?urlpath=proxy%2F1234%2F

I think that's the preferred way to deliver up-to-date mathics for the future. Maybe we can figure out how to point existing docs there.

**Comment by slel** [comment]
The angusgriffith.com certificates have been renewed (thanks @sn6uv for that)
but it would be nice if maintainership of the mathics.org and mathics.net
domain names and hosting could be handed over to the new maintainer,
and if the mathics pages on angusgriffith.com could be made to redirect
or at least point to maintained pages.

The following pages have Mathics or IMathics links that are outdated or
otherwise non-working ("403 forbidden", "This site can’t be reached"...).

- [ ] the page for the "mathics" tag on Stack Overflow
  https://stackoverflow.com/tags/mathics/info
- [ ] the "list of kernels" page on the Jupyter wiki:
  https://github.com/jupyter/jupyter/wiki/Jupyter-kernels

**Comment by slel** [comment]
See also https://github.com/mathics/Mathics/issues/768#issuecomment-490428934.

**Comment by mmatera** [comment]
About Jupyter support, if some pull request from my are accepted, mathics
could partially work with iwolfram. In (Southamerican) summer, I plan to
continue improving it.


Dr. Juan Mauricio Matera

CONICET
IFLP - Dto. de Física UNLP,
Casilla de Correos 67
Calle 49 y 115 s/n,
1900 La Plata, Argentina
Tel +54 221 4246062 (int 294)
Fax +54 221 4236335
e-Mail: matera@fisica.unlp.edu.ar
homepage: http://www.fisica.unlp.edu.ar/Members/matera
Skype: mauricio5831

**Comment by wolfv** [comment]
Hi @mmatera I merged one of your PR's yesterday. Please point me to others you need to have merged.

**Comment by rocky** [comment]
@GarkGarcia I think this should be in milestone 1.1 - any release should have accurate documentation for that release. 

Also fixing this seems pretty straightforward.

**Comment by GarkGarcia** [comment]
I'm not sure how to deal with this, since the is no website running a recent version of Mathics currently online. I think you have to simply remove the link.

**Comment by rocky** [comment]
Is this about having recent docuemntation online? That is what I took the title to mean. 

After the docs build, then just copy them over to https://github.com/mathics/mathics.github.io/tree/master/docs

It will then appear in as http://mathics.github.io/docs/mathics-1.1-dev.pdf

At any rate that's what I took this issue to mean. However if there's more to it let me know.

**Comment by slel** [comment]
Related to this issue:

- Could @poeschko transfer ownership of the mathics.org
  and mathics.net to current maintainers of Mathics?
  Or at least make them redirect to mathics.github.io?

- Could @sn6uv make mathics.angusgriffith.com redirect
  to https://mathics.github.io?

- Could the http version of mathics.github.io be made
  to redirect to the https version?

**Comment by GarkGarcia** [comment]
> Is this about having recent docuemntation online? That is what I took the title to mean.

No, I don't think so. In my understand, this is about links in our documentation that points to mathics.net and mathics.org, which were using outdated versions of Mathics (this websites aren't even up anymore).

> * Could @poeschko transfer ownership of the mathics.org and mathics.net domain names to current Mathics maintainers? Or at least make them redirect to http://mathics.github.io?

That would be great too. However, we'd still need to host the website somewhere. No sure how we'd pay for that, or if anyone would be welling to maintain it.

**Comment by rocky** [comment]
> That would be great too. However, we'd still need to host the website somewhere. No sure how we'd pay for that, or if anyone would be welling to maintain it.

Set up [github sponsors](https://github.com/sponsors). And ask cloud providors like DigitalOcean or some up-and-coming hosting service if they are interested.  

For now, though we should remove that option.

**Comment by GarkGarcia** [comment]
> And ask cloud providors like DigitalOcean or some up-and-coming hosting service if they are interested.

I was thinking about that too. We could try to setup a sponsored instance of Mathics of sorts. A websites that let's people use Mathics on the web and is payed by a company that in turn is allowed to place adds in the site. Something similar to https://haveibeenpwned.com/. I'll investigate this in the future.

> Set up github sponsors.

One issue with that is "who will handle the money?". I'm not sure we could trust anyone with that. That could work if we could somehow make sure that the funds are only spent on hosting bills, but I don't know how to do that.

**Comment by rocky** [comment]
> As mentioned elsewhere, I've set up Mathics in Binder: https://mybinder.org/v2/gh/wolfv/Mathics/binderurlpath=proxy%2F1234%2F

@wolfv Would there be an advantange to setting this up the this github?  

@GarkGarcia Now that I understand this better, I think this should *not* be addressed before the 1.1 release. This will probably be too difficult to do in the short term. 

I thought this was simply about updating the PDF and HTML/XML based docs, and that has been done. Hooking Django showing sections of this I think can be done simply as well and shoudl be for 1.1

Here's why *after* 1.1 makes sense. With everything else more or less stable, The packagers who want to package this for distro can do that. @mmatera has said he would work on a iwolfram interface in a couple of months maybe. Having a solid release to work off of for him and others who want to do similar sorts of things is good.

**Comment by mmatera** [comment]
Binder would be a good option after we have a stable jupyter interface. I could work more in iwolfram maintenance on my holidays during January. Any progress we get in improving the basic support on the text interface of mathics will make my dutty easier...

**Comment by rocky** [comment]
> Any progress we get in improving the basic support on the text interface of mathics will make my dutty easier...

What exactly does this mean. Be as specific as you can. Thanks.

**Comment by mmatera** [comment]
My minimal wish-list would be 
* Make the starting time of mathics as fast as possible.
* Improve as much as possible the performance.
* Fix pymathics in order to load modules for things like "manipulate".
* Give support to export graphics in png format.

**Comment by rocky** [comment]
> * Make the starting time of mathics as fast as possible.
> * Improve as much as possible the performance.

This is vague and we don't really know what's possible. How do you know when the goal is achieved? How long is this going to take? 

Why would this be useful for 1.1 as oppsed to 1.2?  Is the speed right now currently *worse* than 1.0? 

Note that if 1.1 is released, 1.2 could come shortly after 1.1. So what is it about this that makes it important for 1.1 given that there is already a huge amount of stuff that most do not have access to unless they install from source, and OS packagers aren't likely to install. 

There are specific PR's  to address specific items like the #633 and #619 and those seem easy to address.

* Give support to export graphics in png format.

Unless this is easily doable, I suspect in the release after 1.1 this will be addressed by virtue of offloading the work to mathplot via sympy or something else.

**Comment by mmatera** [comment]
* the speed is a little bit worse now than in the previous version, just because we have increased the site of the defintions database.
In that sense, we have already a couple of PRs, including those that you mention,  that seems to help a lot on this aspect.

* About graphics, as you say, I think that using other libraries would be the easier way. 

I guess now that what you want is a list of PRs to fix and incorporate in 1.1 and 1.2 instead of general goals. I could examinate what is now available a make that list during the next days.

**Comment by rocky** [comment]
Thanks for the clarification.

>  I could examinate what is now available a make that list during the next days.

I think that would help. @GarkGarcia ? 

 With respect to 1.1 and 1.2 I am just trying to make concrete what right now seems nebulous so we could organize things and get a sense of where we are. For me, my understanding of what's possible and what is involved has changed in understanding the code better. 

My own personal view is that releasing early and often is great. Releasing broken code though is worse than doing nothing. It is subjective as to when something is "broken" as opposed to to be implemented later. 

As examples, not having a manual (when there was one) in my opinion is not acceptable. B ut not having an online way to run is okay since that's largely broken anyway. So although that is a regression, in a sense it is an okay one.

**Comment by GarkGarcia** [comment]
> @GarkGarcia Now that I understand this better, I think this should not be addressed before the 1.1 release. This will probably be too difficult to do in the short term.

I agree. I didn't plan to contact any companies before the 1.1 release. However, I plan to remove the references to mathics.net and mathics.org from the documentation. We could re-add the links later if we ever get the sites up-and-running again.

**Comment by GarkGarcia** [comment]
> I think that would help. @GarkGarcia ?

I agree, that would help a lot.

On that note, I think we should avoid working on functionality that isn't listed in the 1.1 milestone. While adding functionality is generally something we need to do in the long term (lots of things aren't implemented yet), we need to get the issues of the 1.1 milestone addressed until the start of november. Adding more stuff is likely to introduce bugs, so I'd like to avoid that as much as possible for the moment.

My understanding is that not all of us agree on what should be addressed before the 1.1 release. @rocky @mmatera I think we should try to arrange a meeting (remotely, of course) for us to come up with a list of issues that should be addressed before 1.1, what do you guys think about that? Anyway, my point is: let's make a list of things that should be fixed before 1.1 and after that let's commit on fixing those issues and those issues alone.

**Comment by poeschko** [comment]
I'm happy to point mathics.net and mathics.org somewhere else, just let me know where.

**Comment by mmatera** [comment]
> My understanding is that not all of us agree on what should be addressed before the 1.1 release. @rocky @mmatera I think we should try to arrange a meeting (remotely, of course) for us to come up with a list of issues that should be addressed before 1.1, what do you guys think about that? Anyway, my point is: let's make a list of things that should be fixed before 1.1 and after that let's commit on fixing those issues and those issues alone.

It would be nice. Just in case, I am in Argentina. Where are you? (that is needed to coordinate taking into account the time zone).
then we could arrange a meet at https://meet.jit.si/Mathics1.1milestone (for example)

**Comment by rocky** [comment]
NYC

**Comment by mmatera** [comment]
@GarkGarcia @rocky, can we set the meeting for tomorrow, let's say 10.00 am (Argentina/GTM-3) / 09.00 am (NYC/GTM-4)
 at [](https://meet.jit.si/Mathics1.1milestone)?

**Comment by rocky** [comment]
During the workday week is not possible for me. Pick sometime on the weekend. Thanks.

**Comment by mmatera** [comment]
Next Saturday, same bat-time, same bat-channel, then?

**Comment by slel** [comment]
> I'm happy to point mathics.net and mathics.org somewhere else, just let me know where.

Thanks @poeschko for offering to do that!

From what I understand, mathics dot org was the project homepage
while mathics dot net was a "try Mathics online" website.

The last archived version of both that I could find are:

- [mathics dot org archived on 2018-06-28](http://web.archive.org/web/20180628092428/http://www.mathics.org:80/)
- [mathics dot net archived on 2016-06-01](http://web.archive.org/web/20160601052313/http://www.mathics.net:80/)

For now, in my opinion, what would make the most sense
is to point both to the current Mathics website:

-  http://mathics.github.io

since it can be edited by the current maintainers via the repository at

- https://github.com/mathics/mathics.github.io/

and it will be easy to add a link there on how to use Mathics online.

**Comment by slel** [comment]
Regarding the "try Mathics online" aspect, two things could help:

- #915: make Mathics [Binder ready](https://mybinder.org) and add a Binder button
- #916 : make Mathics work on [CoCalc](https://cocalc.com/index.html) and reopen https://github.com/sagemathinc/cocalc/issues/417

With any (or ideally both) of these, anyone could easily use Mathics online.

**Comment by rocky** [comment]
> With any (or ideally both) of these, anyone could easily use Mathics online.

Yes, but not anyone can make it visable via https://github.com/mathics/mathics.github.io/  We're waiting on that. 

Independent of that currently master is intermittently broken with the Django 2.x upgrade, so we'd need to do something about that or be careful with the version Binder or CoCalc picks up. (We are also waiting for master not to be intermittently broken. See #905

**Comment by GarkGarcia** [comment]
> It would be nice. Just in case, I am in Argentina. Where are you? (that is needed to coordinate taking into account the time zone).
> then we could arrange a meet at https://meet.jit.si/Mathics1.1milestone (for example)

Sorry for the delay, I've been very busy lately 😅️

I'm in Brazil (UTC -3), which I think is the same timezone as yours and @rocky's (according to this random website: https://www.timeanddate.com/time/map/).

**Comment by GarkGarcia** [comment]
> Next Saturday, same bat-time, same bat-channel, then?

That works fine to me!

**Comment by GarkGarcia** [comment]
> For now, in my opinion, what would make the most sense is to point both to the current Mathics website

I'm inclined to that indeed. I'll try to work on it this week.

**Comment by rocky** [comment]
@GarkGarcia - We missed you in the meeting this morning. 

One of the upshots of where things stand now is that because neither of us has access to commit either to this repository or the matihics.github.com repository, and access to both of these are needed as a process thing going forward and for release, we have created a new organization Mathics3.org . (The 3 is for Python3). You have been invited to that organization.

If there are others who would like to join the organization let us know  and we'll add you.

**Comment by rocky** [comment]
Here are my notes on what we discussed earlier: 

What is needed for next release:

- Django 2.x
- Docs (just a pass over the existing documentation, not online evautation)
- The usual PyPI packaging stuff. 
- Access to things that make a release happen (see http://github.com/Mathics3):

I don't think better formatting is needed. It needs to be rethought.

I *would* like to see small stuff in:

* SansSerif #908
* Poke1024's #633 (Faster startup) ROCKY
* Poke1024's #580 (cache for do_format) mmatera
* Poke1024's #619 Fix quadratic runtimes in Mathics (mmatera branch)

Also discussed
*  pymathics module incomleteness problems
* `Import@` vs `<< "https://..."

**Comment by GarkGarcia** [comment]
> @GarkGarcia - We missed you in the meeting this morning.
> 
> One of the upshots of where things stand now is that because neither of us has access to commit either to this repository or the matihics.github.io repository, and access to both of these are needed as a process thing going forward and for release, we have created a new organization Mathics3.org . (The 3 is for Python3). You have been invited to that organization.
> 
> If there are others who would like to join the organization let us know and we'll add you.

I'm sorry for not coming today. I'd be happy to arrange another meeting if you guys are available some other time this weekend. I'm available today.

> One of the upshots of where things stand now is that because neither of us has access to commit either to this repository or the matihics.github.io repository, and access to both of these are needed as a process thing going forward and for release, we have created a new organization Mathics3.org . (The 3 is for Python3). You have been invited to that organization.

Yeah, this is a real problem indeed. @wolfv and @poeschko are members of the Mathics organization, I think they have the authority to invite us to becomes members too. I think creating another organization isn't the most ideal path, but I understand why you guys felt the need for doing so. I'd like to keep our efforts focused in this repository, but this project can't go forward if the people maintaining it don't have the authority necessary to release the code. As such, I'll accept the invitation to be a part of this new organization, but I'd prefer if the organizations eventually got merged.

**Comment by GarkGarcia** [comment]
> * Django 2.x

This is the most relevant issue to me at the moment. @mmatera @rocky Do any of you have experience with Django? As you know, I tried updating to Django 2.2 a while ago and it didn't work out.

**Comment by GarkGarcia** [comment]
> I don't think better formatting is needed. It needs to be rethought.

I agree, this is much more complex than what I originally thought. I'll remove it from the milistone.

**Comment by suhr** [comment]
> If there are others who would like to join the organization let us know and we'll add you.

I would like to.

**Comment by GarkGarcia** [comment]
> * SansSerif #908
> * Poke1024's #633 (Faster startup) ROCKY
> * Poke1024's #580 (cache for do_format) mmatera
> * Poke1024's #619 Fix quadratic runtimes in Mathics (mmatera branch)

I didn't merge this PRs because I was trying to avoid introducing bugs, but sure, I can merge them.

**Comment by mmatera** [comment]
> Yeah, this is a real problem indeed. @wolfv and @poeschko are members of the Mathics organization, I think they have the authority to invite us to becomes members too. I think creating another organization isn't the most ideal path, but I understand why you guys felt the need for doing so. I'd like to keep our efforts focused in this repository, but this project can't go forward if the people maintaining it don't have the authority necessary to release the code. As such, I'll accept the invitation to be a part of this new organization, but I'd prefer if the organizations eventually got merged.

@GarkGarcia, the idea is -at least by now- to have a common space to work together in PRs without breaking the main project. If once the project is ready for release,  we do not have an answer from the owners of this project, we could analyze to publish the release from there.

**Comment by rocky** [comment]
> > If there are others who would like to join the organization let us know and we'll add you.
> 
> I would like to.

@suhr You should have an invitation now.

**Comment by suhr** [comment]
Yeah, thanks.

**Comment by mmatera** [comment]
> I'm sorry for not coming today. I'd be happy to arrange another meeting if you guys are available some other time this weekend. I'm available today.

I will be a little bit busy the rest of the weekend, but if you want to talk a little bit, I could be available for another short meeting.

**Comment by rocky** [comment]
> > Yeah, this is a real problem indeed. @wolfv and @poeschko are members of the Mathics organization, I think they have the authority to invite us to becomes members too. I think creating another organization isn't the most ideal path, but I understand why you guys felt the need for doing so. I'd like to keep our efforts focused in this repository, but this project can't go forward if the people maintaining it don't have the authority necessary to release the code. As such, I'll accept the invitation to be a part of this new organization, but I'd prefer if the organizations eventually got merged.
> 
> @GarkGarcia, the idea is -at least by now- to have a common space to work together in PRs without breaking the main project. If once the project is ready for release, we do not have an answer from the owners of this project, we could analyze to publish the release from there.

And as for day-to-day work - I would prefer making PR's reviews and commits there since we now all have equal access. 

It is possible to issue PRs from there back into Mathics. Whether we do that or how often it is done or if the project owners even *want* that, it is undecided and up for discussion.

**Comment by rocky** [comment]
> > * Django 2.x
> 
> This is the most relevant issue to me at the moment. @mmatera @rocky Do any of you have experience with Django? As you know, I tried updating to Django 2.2 a while ago and it didn't work out.

I don't have experience with Django 2.2 or else I would have helped and it is why I posted something on the Django forum. 

However if this isn't solved by the time other things are ready and is holding us back, I'll dive into this. I have worked with other web frameworks, so how diffucult could this be?

**Comment by mmatera** [comment]
OK, I agree that we want  Django 2.2 working before 1.1 release, but let face it when at least the mathics kernel works well. Then, the problems with the graphic interface would not be originated from problems with the kernel.

**Comment by rocky** [comment]
> 
> I'm sorry for not coming today. I'd be happy to arrange another meeting if you guys are available some other time this weekend. I'm available today.
> 
'
I can't this weekend, but we can plan on another one next weekend. Suggest a time.

**Comment by GarkGarcia** [comment]
> > I'm sorry for not coming today. I'd be happy to arrange another meeting if you guys are available some other time this weekend. I'm available today.
> 
> '
> I can't this weekend, but we can plan on another one next weekend. Suggest a time.

What about next Saturday at 5 pm in the UTC-3 timezone?

**Comment by GarkGarcia** [comment]
> OK, I agree that we want Django 2.2 working before 1.1 release, but let face it when at least the mathics kernel works well. Then, the problems with the graphic interface would not be originated from problems with the kernel.

What would you like to get working before tackling the Django issue? In my opinion fixing graphics is something that will take a lot of time and that can't be properly addressed before november.

**Comment by GarkGarcia** [comment]
> > > Yeah, this is a real problem indeed. @wolfv and @poeschko are members of the Mathics organization, I think they have the authority to invite us to becomes members too. I think creating another organization isn't the most ideal path, but I understand why you guys felt the need for doing so. I'd like to keep our efforts focused in this repository, but this project can't go forward if the people maintaining it don't have the authority necessary to release the code. As such, I'll accept the invitation to be a part of this new organization, but I'd prefer if the organizations eventually got merged.
> > 
> > 
> > @GarkGarcia, the idea is -at least by now- to have a common space to work together in PRs without breaking the main project. If once the project is ready for release, we do not have an answer from the owners of this project, we could analyze to publish the release from there.
> 
> And as for day-to-day work - I would prefer making PR's reviews and commits in Mathics3, since we now all have equal access.
> 
> It is possible to issue PRs from there back into Mathics. Whether we do that or how often it is done or whether the project owners even _want_ that, it is undecided and up for discussion. (Silence will be taken to mean go with the other organization.)

Fair enough. It would be cool @wolfv or @poeschko could make us part of the Mathics organization as soon as possible, but if they can't I'll start pushing my commits to the Mathics3 fork.

> @GarkGarcia, the idea is -at least by now- to have a common space to work together in PRs without breaking the main project. If once the project is ready for release, we do not have an answer from the owners of this project, we could analyze to publish the release from there.

Makes sense too. I don't see how we could resolve our governance issues in here without the help of someone who is part of the Mathics organization (so we don't have another option, at least until someone invites us to be a part of the organization).

**Comment by GarkGarcia** [comment]
This issue has been mostly addressed by https://github.com/mathics/Mathics/pull/921. @mmatera @rocky I guess we can close the issue?

We really shouldn't discuss our governance issues in here. Let's continue the discussion in https://github.com/mathics/Mathics/issues/841.

**Comment by rocky** [comment]
> What would you like to get working before tackling the Django issue? 

For myself, it is just a matter of seeing that more or less everything else that is easily covered is covered and that the main thing blocking us is having this resolved.

I may only have for a while short amounts of time to work on this. So for me and right now I have this backlog of little stuff that had been accumulating that I'd like to sort out first.

**Comment by rocky** [comment]
> This issue has been mostly addressed by #921. @mmatera @rocky I guess we can close the issue?

This would have been fine if we any of us had access to mathics.github.io. The the final bit about going to http://mathics.github.io/ and seeing somethat is current and works would be solved. Right now that that is not the case. 

We _can_ make this work using the new organization in short order though. 

As for the process about closing the issue (even though in my opinion right now this is still broken) is up to you. As for the mechanics of closing this and opening another or however you want to handle this "Issue"-wise I'll leave up to you.

**Comment by mmatera** [comment]
> > > I'm sorry for not coming today. I'd be happy to arrange another meeting if you guys are available some other time this weekend. I'm available today.
> > 
> > 
> > '
> > I can't this weekend, but we can plan on another one next weekend. Suggest a time.
> 
> What about next Saturday at 5 pm in the UTC-3 timezone?

OK. Let's schedule then a new meeting  on https://meet.jit.si/Mathics1.1milestone  for  October 3th, 17.00hs (GTM-3)

**Comment by rocky** [comment]
I think I can make it but if I can't I'll let you know.

**Comment by poeschko** [comment]
@GarkGarcia @rocky I added you to a new team @mathics/maintainers which has Maintainer access to the repos Mathics and mathics.github.io. Let me know if you need anything beyond that.

Regarding domains, I'll point mathics.org to https://mathics.github.io – any suggestions what should happen with mathics.net?

**Comment by mmatera** [comment]
In any case, maybe it would be a good idea to work on the "fork"  organization, in a way to do not break the main repository. Then we can merge everything here.

**Comment by slel** [comment]
Suggestion: point mathics.net to https://mathics.gighub.io/try
and put some words there explaining ways to try mathics online.
If we find a way to revive something like mathics.net we can
either make it live at that "try" page or at least link from there.

**Comment by rocky** [comment]
Many thanks!

I will be testing this out. The first order of business will be to try to
add a PDF for the 1.1 dev branch.

**Comment by rocky** [comment]
> Suggestion: point mathics.net to https://mathics.gighub.io/try and put some words there explaining ways to try mathics online. If we find a way to revive something like mathics.net we can either make it live at that "try" page or at least link from there.

@selel I am not exactly sure what you mean here. Would you fork the code and put in a PR for this. If I have access, I'll merge that. Thanks.

**Comment by rocky** [comment]
@poeschko Many thanks for doing this. Would you also consider adding @mmatera ?

 I have just updated  https://mathics.github.io so that it no longer has broken links.

**Comment by poeschko** [comment]
Yes, of course. I added @mmatera to @mathics/maintainers.

I'm happy to see some increased interest in Mathics again!

It would be nice if mathics.net was a place where you could directly use it, not just a description of how to do it (that would go on mathics.org). If you manage to set that up somewhere, let me know and I'll change its DNS.

**Comment by rocky** [comment]
Absolultey! Will do.

>It would be nice if mathics.net was a place where you could directly use it, not just a description of how to do it (that would go on mathics.org). If you manage to set that up somewhere, let me know and I'll change its DNS.

Right now it feels for me like we are fighting too many small fires. #916 is a plan for doing this. Hopefully someone will volunteer in the short term to make it happen.

**Comment by slel** [comment]
> It would be nice if mathics.net was a place where you could directly use it,
> not just a description of how to do it (that would go on mathics.org).
> If you manage to set that up somewhere, let me know and I'll change its DNS.

Can you describe how you had it set up at mathics.net when it was active?
With a description I can ask my department or university about hosting this.

**Comment by GarkGarcia** [comment]
> > This issue has been mostly addressed by #921. @mmatera @rocky I guess we can close the issue?
> 
> This would have been fine if we any of us had access to mathics.github.io. The the final bit about going to http://mathics.github.io/ and seeing somethat is current and works would be solved quickly. Right now that that is not the case.
> 
> We _can_ make this work using the new organization in short order though.
> 
> As for the process about closing the issue (even though in my opinion right now this is still broken) is up to you.

Fair enough. I'll update the information in mathics.github.io and then I'll close the issue. Does that work to you?

**Comment by GarkGarcia** [comment]
> Regarding domains, I'll point mathics.org to https://mathics.github.io

Actually, I think using the mathics.org domain for mathics.github.io would be nicer, don't you guys agree? GithubPages allows users to use custom domains.

> any suggestions what should happen with mathics.net?

For now I think our best option is to point it to mathics.org or mathics.github.io. We can set it up again if we ver menage to get it running as before.

**Comment by GarkGarcia** [comment]
> I have just updated https://mathics.github.io so that it no longer has broken links.

Great! Thanks! I guess we can close this issue then?

**Comment by GarkGarcia** [comment]
> Can you describe how you had it set up at mathics.net when it was active?
> With a description I can ask my department or university about hosting this.

That would be amazing!

**Comment by rocky** [comment]
> I guess we can close this issue then?

This issue is kind of a tough one with many levels. The top link seems to refer to the online-help. As far as I can tell that reliase on the Javascript `$doc` variable getting set and I have no idea what that *should* be. I downloaded the 1.0 tarball from PyPI and I don't see docs in that. 

When I currenlty open `mathicsserver` and go to docs I see blank info, not obsolete info. 

So in a sense, yes, that is fixed. Maybe a "better" fix would be to disable the doc ability altogether. I don't know. 

Again this is the data, I leave to you what you want to do with it or whether you want to close this issue.

**Comment by GarkGarcia** [comment]
Also, @rocky @mmatera I guess the other organization is kinda irrelevant now? Is a reason to keep it?

**Comment by poeschko** [comment]
I also updated the repo details of https://github.com/mathics/Mathics so it points to https://mathics.org (again), since that's equivalent to (and the canonical place for) https://mathics.github.io now.

> Can you describe how you had it set up at mathics.net when it was active?

That was a while ago and that server is long gone, unfortunately... I *think* it was a fairly standard Django webserver. The biggest complication was getting Sage to work on the server, but that's no longer a required dependency for Mathics, anyway (right?). It did support logging in and saving "notebooks" (using a MySQL database). As far as I remember, there was a way to turn off any file-related operations (or other potentially unsafe operations) in the Mathics interpreter, and that was basically the security mechanism.

There was definitely a stability issue, so every now and then (perhaps if somebody evaluated something "special"), the server became unresponsive. Maybe this is not an issue anymore with all the technologies involved (most importantly the Mathics code base) having evolved in the meanwhile, but it's certainly something to watch out for.

Good luck! :-)

**Comment by rocky** [comment]
> I guess the other organization is kinda irrelevant now? Is a reason to keep it?

Yes. Although I have removed whatever is exists here, it appears that I can't create *new* repositories under Mathics and I assume @mmatera can't either. I have a new repository for splitting out a terminal CLI. mmatera added one from his namespace as well. 

Also in the future recall we want separate Python modules for `LoadModule` imports.  Specificaly one for natural language stuff, one for graphs, and so on. This would naturally go right under Mathics. 

@poeschko if you could make that happen to I would be eternally grateful and will remove the Organiation as soon as we copy the remaining repositories to Mathics. Many thanks in advance.

**Comment by GarkGarcia** [comment]
> > I guess we can close this issue then?
> 
> This issue is kind of a tough one with many levels. The top link seems to refer to the online-help. As far as I can tell, that relies on the Javascript `$doc` variable getting set and I have no idea what that _should_ be. I downloaded the 1.0 tarball from PyPI and I don't see docs in that.
> 
> When I currenlty open `mathicsserver` and go to docs I see blank info, not obsolete info.
> 
> So in a sense, yes, that is fixed. Maybe a "better" fix would be to disable the doc ability altogether. I don't know.
> 
> Again this is the data, I leave to you what you want to do with it or whether you want to close this issue.

Humm, I see. Indeed there are still references to mathics.net in the code (something related to authentication) and I don't understand how the "online help" stuff you mentioned worked. I'll keep the issue opened while we figure this things out.

**Comment by rocky** [comment]
> As far as I remember, there was a way to turn off any file-related operations (or other potentially unsafe operations) in the Mathics interpreter, and that was basically the security mechanism.

Alas I recently made this *worse* by adding `ProcessId`, `ParentProcessId`, `MachineId` and so on. But assuming you get everything *else* worked out, I'd be happy to figure out a hack for removing these functions. 

Probalby there might be an attribute added that indicates these are not suitable for online live servers.

**Comment by GarkGarcia** [comment]
> > As far as I remember, there was a way to turn off any file-related operations (or other potentially unsafe operations) in the Mathics interpreter, and that was basically the security mechanism.
> 
> Alas I recently made this _worse_ by adding `ProcessId`, `ParentProcessId`, `MachineId` and so on. But assuming you get everything _else_ worked out, I'd be happy to figure out a hack for removing these functions.
> 
> Probalby there might be an attribute added that indicates these are not suitable for online live servers.

That's a great idea! We should probably add a switch to disable IO functionalities altogether (reading an writing to disk, but only the "user-controled" stuff, like saving images to disk).

**Comment by poeschko** [comment]
Shall we move the discussion about new repositories to https://github.com/orgs/mathics/teams/maintainers/discussions? Let me know there what repositories you'd like to have created.

**Comment by GarkGarcia** [comment]
> Shall we move the discussion about new repositories to https://github.com/orgs/mathics/teams/maintainers/discussions? Let me know there what repositories you'd like to have created.

Yes! I try to make a summery of our current discussion in there.

**Comment by GarkGarcia** [comment]
> > Shall we move the discussion about new repositories to https://github.com/orgs/mathics/teams/maintainers/discussions? Let me know there what repositories you'd like to have created.
> 
> Yes! I try to make a summery of our current discussion in there.

See https://github.com/orgs/mathics/teams/maintainers/discussions/3 for the summery.

**Comment by rocky** [comment]
I just spent some time with git bisect to figure out which commit broke the docs. It appears to be: 

```
ee1677d1a528ce1633109b59674d08fb1c62eede is the first bad commit
commit ee1677d1a528ce1633109b59674d08fb1c62eede
Author: Angus Griffith <16sn6uv@gmail.com>
Date:   Sun Mar 12 22:32:53 2017 -0700

    basic Django 1.11 support

:040000 040000 79dd6a61dd4e9a948c97a5eef2128255b4f52d9e 1c3028f3560da350c3e272d8fd37c1f8f6cab05f M	mathics
:100644 100644 048e619d9d4763e36681186f69af9efad84a4c3a f6c0463c7783069648fea1ecb5a825d538994204 M	setup.py
```

**Comment by rocky** [comment]
I believe this is now fixed in bc51171fc9b45d56f9aed217e85293b5b42cc4b4 

Of course there's always more to do. but my take is this should be follow-on isssues.

For example going over the content to ensure it has everything. (Im too tired to do right now).

Right now, I am not seeing docs on newly-added functions. 

Where exactly is this information coming from?

**Comment by GarkGarcia** [comment]
> I believe this is now fixed in [bc51171](https://github.com/mathics/Mathics/commit/bc51171fc9b45d56f9aed217e85293b5b42cc4b4)

That's great! Awesome work!
